proposals/p3848.md

```diff
@@ -0,0 +1,576 @@
+# Lambdas
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/3848)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Syntax Overview](#syntax-overview)
+-   [Introducer](#introducer)
+-   [Positional Parameters](#positional-parameters)
+    -   [Positional Parameter Restrictions](#positional-parameter-restrictions)
+-   [Function Captures](#function-captures)
+    -   [Default Capture Mode](#default-capture-mode)
+    -   [Capture Modes](#capture-modes)
+-   [Function Fields](#function-fields)
+-   [Copy Semantics](#copy-semantics)
+-   [Self and Recursion](#self-and-recursion)
+-   [Rationale](#rationale)
+
+<!-- tocstop -->
+
+## Abstract
+
+To support migration from C++ to Carbon, there must be valid syntax to capture
```

Comment:
Stating the rationale solely in terms of C++ migration seems too narrow -- for example, it doesn't explain why we have positional parameters, or the `=>` syntax, which are unlikely to be used for C++ migration purposes. I think we want lambdas because they can make code easier to read, understand, and write, as well as because we need them for migration.

Reply:
Good call, thanks!